specification/cognitiveservices/Language.AnalyzeText/models/common.models.tsp

```diff
@@ -0,0 +1,974 @@
+  summaryLength?: SummaryLengthBucket;
+}
+
+@doc("The context of the summary.")
+model SummaryContext {
+  @doc("Start position for the context. Use of different 'stringIndexType' values can affect the offset returned.")
+  offset: int32;
+
+  @doc("The length of the context. Use of different 'stringIndexType' values can affect the length returned.")
+  length: int32;
+}
+@clientName("CustomTaskContent", "csharp")
+@doc("Parameters object for a text analysis task using custom models.")
+model CustomTaskParameters is TaskParameters {
+  @doc("This field indicates the project name for the model.")
+  projectName: string;
+
+  @doc("This field indicates the deployment name for the model.")
+  deploymentName: string;
+}
+
+@doc("Custom Task Results")
+model CustomResult {
+  @doc("Errors by document id.")
+  errors: Array<DocumentError>;
+
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type"
+  @doc("if showStats=true was specified in the request this field will contain information about the request payload.")
+  statistics?: RequestStatistics;
+
+  @doc("This field indicates the project name for the model.")
+  projectName: string;
+
+  @doc("This field indicates the deployment name for the model.")
+  deploymentName: string;
+}
+
+@doc("Optional parameter to use a Custom Character to be used for redaction in PII responses. Default character will be * as before. We allow specific ascii characters for redaction.")
+enum redactionCharacter {
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  exclamationPoint: "!",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  numberSign: "#",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  dollar: "$",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  perCent: "%",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  ampersand: "&",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  asterisk: "*",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  plus: "+",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  minus: "-",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  equals: "=",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  questionMark: "?",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  atSign: "@",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  caret: "^",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  underscore: "_",
+  #suppress "@azure-tools/typespec-azure-core/documentation-required"
+  tilde: "~",
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@clientName("AnalyzeTextError", "csharp")
+@doc("The error object.")
+model Error is Record<unknown> {
```

Comment:
Why is `Error` `Record<unknown>`?

Reply:
This is the same as the comment above

Reply:
What additional properties will the service ever send back to the client? And why are they not documented/described in the API? There is nothing the client can do with that information, right? Other than at most printing it...

Reply:
We should remove the `Record<unknown>`

Reply:
done.